### ISSUE TITLE
TensorFlow-2.3.1-fosscuda-2019b-Python-3.7.4 build fails

### ISSUE THREAD

**Issue by Jojuh** [easybuilders/easybuild-easyconfigs#12105]
Building TensorFlow 2.3.1 (or 2.2.0) with fosscuda-2019b fails with different error messages.

eb version: `This is EasyBuild 4.3.2 (framework: 4.3.2, easyblocks: 4.3.2)`

` eb -r TensorFlow-2.3.1-fosscuda-2019b-Python-3.7.4.eb` produces a lot of errors and ultimately (the actual compilation target varies):

```
ERROR: /opt/ext/easybuild/build/TensorFlow/2.3.1/fosscuda-2019b-Python-3.7.4/TensorFlow/tensorflow-2.3.1/tensorflow/python/tools/BUILD:179:10 C++ compilation of rule '//tensorflow/core/kernels:fake_quant_ops_gpu' failed (Exit 1): crossto
ol_wrapper_driver_is_not_gcc failed: error executing command...
```
```
ERROR: /opt/ext/easybuild/build/TensorFlow/2.3.1/fosscuda-2019b-Python-3.7.4/TensorFlow/tensorflow-2.3.1/tensorflow/lite/toco/python/BUILD:84:10 C++ compilation of rule '//tensorflow/core/kernels:argmax_op_gpu' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command
```
and
```
ERROR: /opt/ext/easybuild/build/TensorFlow/2.3.1/fosscuda-2019b-Python-3.7.4/TensorFlow/tensorflow-2.3.1/tensorflow/lite/toco/python/BUILD:84:10 C++ compilation of rule '//tensorflow/core/kernels:colorspace_op_gpu' failed (Exit 1): cross
tool_wrapper_driver_is_not_gcc failed: error executing command
```

As the failing build target varies I'm thinking this is due to parallelisation. I tried adding `--cuda-compute-capabilities 7.0` as we are running Tesla V100s but no change. TensorFlow 2.2.0 fails similarly. Building with just foss-2019b succeeds with no errors. Should I try to update system drivers for nvidia? We're using 450.80.02.

**Comment by Jojuh** [comment]
Seems to be related to https://bugzilla.redhat.com/show_bug.cgi?id=1925204.

Downgrading glibc* to 2.17-317 helped and the build completed succesfully.

Closing as this is clearly a system issue.

**Comment by boegel** [comment]
Thanks for following up on this after you figured out the underlying cause!

**Comment by terjekv** [comment]
https://centos.pkgs.org/7/centos-updates-x86_64/glibc-2.17-322.el7_9.x86_64.rpm.html presents the following changelog:

```
2021-01-06 - Carlos O'Donell <carlos@redhat.com> - 2.17-322
- Enable file-based IFUNC selection on NVMe devices (#1883162)
2021-01-06 - Carlos O'Donell <carlos@redhat.com> - 2.17-321
- CVE-2020-10029: Prevent stack corruption from crafted input in cosl, sinl, sincosl, and tanl function. (#1812119)
2021-01-05 - Carlos O'Donell <carlos@redhat.com> - 2.17-320
- CVE-2020-29573: Harden printf family of functions (#1869380)
2021-01-05 - Carlos O'Donell <carlos@redhat.com> - 2.17-319
- Revert fix for #1772307 to improve Intel Xeon performance (#1889977)
2021-01-05 - Carlos O'Donell <carlos@redhat.com> - 2.17-318
- CVE-2019-25013: Fix EUC-KR conversion module defect (#1912543)
2020-05-12 - Florian Weimer <fweimer@redhat.com> - 2.17-317
- Do not clobber errno in nss_compat (#1834816)
[...]
```
3 CVEs, one NVMe-related fix, and one performance tuning for Xeon ([#1889977](https://bugzilla.redhat.com/show_bug.cgi?id=1889977))... That looks interesting! But, alas, after logging in to RedHat's bugzilla, I'm met with this:

```
You are not authorized to access bug #1889977.

Most likely the bug has been restricted for internal development processes and we cannot grant access.

If you are a Red Hat customer with an active subscription, please visit the Red Hat Customer Portal for assistance with your issue

If you are a Fedora Project user and require assistance, please consider using one of the mailing lists we host for the Fedora Project.
```

So... maybe it's relevant?

**Comment by Flamefire** [comment]
FTR: `crosstool_wrapper_driver_is_not_gcc failed: error executing command` just says that the compile failed. The error message WHY it fails is a bit below. That might allow you to google the real issue :)
